### ISSUE TITLE
autofixing object-shorthand when some properties are keywords leads to parse error

### ISSUE THREAD

**Issue by lukeapage** [eslint/eslint#6840]
**What version of ESLint are you using?**
3.1.1
**What parser (default, Babel-ESLint, etc.) are you using?**
default es6
**Please show your full configuration:**
`/* eslint object-shorthand: "error" */`
**What did you do? Please include the actual source code causing the issue.**
```
var a = {
   b() {
      return {
          arguments: arguments,
      };
   }
};
```
**What did you expect to happen?**
either to convert when fixing to `{ arguments }` and not give a parser error or to leave as-is
**What actually happened? Please include the actual, raw output from ESLint.**

it was autofixed and then the next run gave this error
 `error  Parsing error: 'arguments' can not be used as shorthand property`

this comes from acorn here: https://github.com/ternjs/acorn/blob/5e5e4301444d4471b62c9079d7c049afd9598ac0/src/expression.js#L520

I'm not sure if its completely correct that all keywords like arguments should give syntax errors - it doesn't error in chrome, but that doesn't mean anything. So it could be an upstream acorn bug. But it does look like `this` is invalid as a shorthand property definitely and I can't see code in the eslint rule that checks for any keywords.

**Comment by platinumazure** [comment]
Per [the spec](http://www.ecma-international.org/ecma-262/6.0/#sec-reserved-words), I would say we need to check for reserved words for sure.

`arguments` doesn't count as a reserved words (at least in ES2015), so that seems to be a bug in Acorn and/or espree, unless I'm missing something in the spec.

I think we definitely have a bug in terms of at least autofixing too eagerly on sure keywords. I'll label this accordingly.

**Comment by mysticatea** [comment]
I think it's acorn's bug, too.

FWIW, Annex C says it forbids `arguments` on `BindingIdentifier`, but in this case, the `arguments` of `{arguments}` is not `BindingIdentifier`. It's `IdentifierReference`.

> It is a SyntaxError if the IdentifierName eval or the IdentifierName arguments occurs as a BindingIdentifier within strict mode code (12.1.1).
> http://www.ecma-international.org/ecma-262/7.0/#sec-strict-mode-of-ecmascript

I guess [this line](https://github.com/ternjs/acorn/blob/5e5e4301444d4471b62c9079d7c049afd9598ac0/src/expression.js#L521) is extra.

**Comment by michaelficarra** [comment]
There's nothing that needs to be done here.

**Comment by platinumazure** [comment]
I concur with @michaelficarra that the rule does not need to be updated. I just checked the rule and it only checks for Identifier object values (reserved words would not be Identifier nodes).

I don't agree that there's nothing for acorn or esprima to do-- from what I understand of the spec, they should accept an "arguments" object shorthand key/value. @michaelficarra If there is something in the spec that I am missing, which contradicts my claim, please point it out to me.

**Comment by mysticatea** [comment]
I sent a PR: https://github.com/ternjs/acorn/pull/444

**Comment by michaelficarra** [comment]
@platinumazure Yes, that's what I meant. I didn't see why we were keeping this issue open.

**Comment by mysticatea** [comment]
OK. I'll close this issue as no change for eslint.
@lukeapage Please track https://github.com/ternjs/acorn/issues/444 for this.
